crates/biome_js_analyze/src/lint/nursery/use_trim_start_end.rs

```diff
@@ -0,0 +1,137 @@
+use biome_analyze::{
+    context::RuleContext, declare_rule, ActionCategory, Ast, FixKind, Rule, RuleDiagnostic,
+    RuleSource,
+};
+use biome_console::markup;
+use biome_js_factory::make;
+use biome_js_syntax::JsCallExpression;
+use biome_rowan::{AstSeparatedList, BatchMutationExt, TextRange, TokenText};
+
+use crate::JsRuleAction;
+
+declare_rule! {
+    /// Enforce the use of `String.trimStart()` and `String.trimEnd()` over `String.trimLeft()` and `String.trimRight()`.
+    ///
+    /// While `String.trimLeft()` and `String.trimRight()` are aliases for `String.trimStart()` and `String.trimEnd()`,
+    /// only using the latter pair ensures consistency and is preferable for their direction-independent wording.
+    ///
+    /// Note that `String.trimStart()` and `String.trimEnd()` methods do not take any parameters. Any arguments passed to these methods will be ignored.
+    /// See the MDN documentation for more details:
+    /// - [String.prototype.trimStart()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trimStart)
+    /// - [String.prototype.trimEnd()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trimEnd)
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// const foo = bar.trimLeft();
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// const foo = bar.trimRight();
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// const foo = bar.trimStart();
+    /// ```
+    ///
+    /// ```js
+    /// const foo = bar.trimEnd();
+    /// ```
+    ///
+    pub UseTrimStartEnd {
+        version: "next",
+        name: "useTrimStartEnd",
+        language: "js",
+        recommended: false,
+        sources: &[RuleSource::EslintUnicorn("prefer-string-trim-start-end")],
+        fix_kind: FixKind::Safe,
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct UseTrimStartEndState {
+    member_name: TokenText,
+    span: TextRange,
+    replaced_member_name: &'static str,
+}
+
+impl Rule for UseTrimStartEnd {
+    type Query = Ast<JsCallExpression>;
+    type State = UseTrimStartEndState;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let arguments = node.arguments().ok()?;
+        let args = arguments.args();
+
+        if !args.is_empty() {
```

Comment:
I confirmed that `String.trimStart()` / `String.trimEnd()` accept the arguments in browser(Chrome, Firefox) and node.js like below:

```
$ echo 'let str = "   Hello, world!   "; console.log(str.trimStart("hoge")); console.log(str.trimEnd("hoge"));' > test.js
node test.js
```

but the implementation should adhere the specification maybe...? @Sec-ant 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trimStart#parameters
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trimEnd#parameters
https://tc39.es/ecma262/#sec-string.prototype.trimstart
https://tc39.es/ecma262/#sec-string.prototype.trimend

ref: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/test/prefer-string-trim-start-end.mjs#L22-L23

Reply:
In JavaScript, if the function accepts no arguments, we can still pass whatever arguments we want and they are simply ignored. So this is not against the spec.

But I think we can ignore these cases when there're arguments passed, because they may be some custom APIs. We can just focus on cases with no arguments.